### ISSUE TITLE
Remove file from commit

### ISSUE THREAD

**Issue by SAxcell** [forcedotcom/devops-center-feedback#44]
**Describe the bug**
Promoting from Dev to UAT, got a Promotion Failed error due to a couple of report types included that are part of managed packages and can't be modified.  Now I can't find a way to go back and remove those two items and try again.  

**To Reproduce**
From Pipeline, select "Change in Shipping Address Requirements"
Click Promote Selected
Promotion Failed
Details:  
![image](https://user-images.githubusercontent.com/84103048/120698136-9cc3d280-c47c-11eb-9318-47a044f3218e.png)

...

**Expected behavior**
I would hope it didn't fail!  However, knowing why it failed, I expect to be able to correct the issue and try to promote again.  

**Screenshots**

**Additional context**
I assumed I could go back to the Work Item, pull the change again, and not select those two items that are part of managed packages, then try again.  However, I don't seem to have any ability to do that.  Help?

**Comment by franciscoperezsammartino** [comment]
Hi @SAxcell , were you able to get around with this?
Thanks again and sorry for the delay.

**Comment by kfidelak94** [comment]
The planned solution to this is to allow for "Revert Commit" from DevOps Center - to remove/revert a file from a work item after it's been committed and before it's been promoted.

**Comment by scottywong** [comment]
This is an important item for us to help better adopt DevOps Center especially for admins. 

It's been close to a year, is there any chance this is progressing past the "researching" stage?

**Comment by srathjenSFDC** [comment]
Definitely need this! I have to make multiple work items and reselect everything to correct a single promotion error.  :'(

**Comment by kristinecurington** [comment]
We have struggled with this issue.  We had a failed promotion because a list view had been deleted in our sandbox and production, when the work item was promoted, it failed because it was trying to delete the list view that had already been deleted.  It mentioned a "destructiveChanges" file, but I couldn't find it.

**Comment by SE-Sleek** [comment]
Yes please, this! All of these "Never" work items are becoming quite annoying...

**Comment by jack-donnelly-thunder** [comment]
Reverting a commit would be awesome to see!  It'd get a lot of usage!

**Comment by ARAGNADevOps** [comment]
Upvote!

**Comment by oleg-mastriukov** [comment]
If you want to keep changes in your repo but don't want DevOps to try to promote them through the pipeline, you can delete the record related to your WI. As a result, the changes will be merged to the destination brunch, but DevOps won't deploy them to the destination org. Use this SOQL and delete a record that refers to the changes you don't want to be deployed to the destination org:
`select fields(all) from sf_devops__Change_Submission__c where sf_devops__Work_Item__c ='your_WI_id' limit 200`

Also, each sf_devops__Change_Submission__c  record has related sf_devops__Submit_Component__c records that contain a list of components to be deployed/deleted. You can delete these records do prevent particular components from being deployed to or deleted from the destination org. Here's an example of a destructive change:
`SELECT Id, sf_devops__Change_Submission__r.sf_devops__Work_Item__r.Name , sf_devops__File_Path__c, sf_devops__Operation__c, sf_devops__Source_Component__c FROM sf_devops__Submit_Component__c
WHERE sf_devops__Operation__c = 'REMOVE'
AND sf_devops__Change_Submission__r.sf_devops__Work_Item__r.Id = 'your_WI_id'`

**Comment by qdmistry** [comment]
If I connect my devops' github repo via [GitButler](https://gitbutler.com/), would I be able to utilize the GitButler's features to roll back, removing file from commit etc or not possible as DevOps cannot update similar to GitButler? Has anyone tried it? 

Impact to other stories on the board:
- [Rollback File](https://github.com/orgs/forcedotcom/projects/4/views/1?pane=issue&itemId=11922600)
- [Hotfix use case](https://github.com/orgs/forcedotcom/projects/4/views/1?pane=issue&itemId=11922622)

**Comment by Swantzify** [comment]
Another big one that is needed ASAP!

**Comment by SarikaPfizer** [comment]
This feature is really needed. We have lot of scenarios where few of the components are not approved for release and we want to remove them from work item from UAT stage so that they can be available in UAT sandbox but not get promoted to next stages. 
Even if we create a new work item from dev stage with final component list it has lot of limitations i.e
1.  components won't be available for commit as those are already upto date in branch. We have to add some dummy change so that it will be detected as new commit. 
2. Somehow if we manage to commit those and deploy till uat devops center in backend merge uat changes with next stage and unwanted changes gets deployed to next stage.

**Comment by KemalPreljevic** [comment]
Any Updates on this Feature? 

I know the manual work around but this is tedious work and a time waster

**Comment by AlbanDugueOrnikar** [comment]
Upvote pleaase team 🙏💖 this is a must have for us to have a better experience

**Comment by shanonlebreton7** [comment]
much needed !! 🙏

**Comment by svinuesa** [comment]
Upvote

**Comment by ChrisITBRM** [comment]
Upvote

**Comment by phillipposkus** [comment]
Upvote!
